.github/workflows/ci.yml

```diff
@@ -0,0 +1,164 @@
+name: CI
+
+on:
+  pull_request:
+  push:
+    branches: [main]
+
+jobs:
+  pre-commit:
+    name: Pre-commit
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v3
+      - uses: pre-commit/action@v3.0.0
+
+  # unit-tests:
```

Comment:
Why is this commented out?

Reply:
Because we haven't written the unit tests as of now and @suryabulusu suggessted to just make a temp setup for now